core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -848,29 +849,30 @@ class QualificationAppInfo(
    * and extracts executor and driver instance types (databricks only)
    */
   override def buildClusterInfo: Unit = {
-    // TODO: Handle dynamic allocation when determining the number of nodes.
-    sparkProperties.get("spark.dynamicAllocation.enabled").foreach { value =>
-      if (value.toBoolean) {
-        logWarning(s"Application $appId: Dynamic allocation is not supported. " +
-          s"Cluster information may be inaccurate.")
-      }
-    }
     // try to figure out number of executors per node based on the executor info
     // Group by host name, find max executors per host
     val execsPerNodeList = executorIdToInfo.values.groupBy(_.host).mapValues(_.size).values
-    val numExecsPerNode = execsPerNodeList.reduceOption(_ max _).getOrElse(0)
-    val activeExecInfo = executorIdToInfo.values.collect {
-      case execInfo if execInfo.isActive => (execInfo.host, execInfo.totalCores)
+    // if we have different number of execs per node, then we blank it out to indicate
+    // not applicable (like when dynamic allocation is on in multi-tenant cluster)
+    // Since with dynamic allocation you could end up with more executors on a node then it
+    // has slots, just always set numExecsPerNode to -1 when its enabled.
+    val dynamicAllocEnabled = Platform.isDynamicAllocationEnabled(sparkProperties)
+    val numExecsPerNode = if (!dynamicAllocEnabled && execsPerNodeList.size > 0 &&
+      execsPerNodeList.forall(_ == execsPerNodeList.head)) {
+      execsPerNodeList.head
+    } else {
+      -1
     }
-    if (activeExecInfo.nonEmpty) {
-      val (activeHosts, coresPerExecutor) = activeExecInfo.unzip
-      if (coresPerExecutor.toSet.size != 1) {
+    val execCoreCounts = executorIdToInfo.values.map(_.totalCores)
+    if (execCoreCounts.size > 0) {
+      if (execCoreCounts.toSet.size != 1) {
         logWarning(s"Application $appId: Cluster with variable executor cores detected. " +
           s"Using maximum value.")
       }
       // Create cluster information based on platform type
-      pluginTypeChecker.platform.configureClusterInfoFromEventLog(coresPerExecutor.max,
-        numExecsPerNode, activeHosts.toSet.size, sparkProperties, systemProperties)
+      platform.configureClusterInfoFromEventLog(execCoreCounts.max,
+        numExecsPerNode, maxNumExecutorsRunning, maxNumNodesRunning,
+        sparkProperties, systemProperties)
     } else {
       // if no executors do we want to qualify at all?  maybe not, else we could look at
```

Comment:
The warning message in the else block should be updated, as the if condition does not check for active executors. 

Infact, should we throw an exception in the else block? This would indicate that no executors were found.